### ISSUE TITLE
DNS challenge fails for `cloudflare` provider.

### ISSUE THREAD

**Issue by A-Shleifman** [containous/traefik#3761]
### Do you want to request a *feature* or report a *bug*?

BUG

### What did you do?

I'm running traefik in a docker container based on traefik:1.7-alpine. The actual version is v1.7.0-rc3.
When I use `route53` provider, traefik retrieves certificates consistently, however, it fails to get certificates for `cloudflare` domains most of the time.
Four out of five times I got `incorrect TXT record` error.

That's what @nmengin wrote about this problem:

```
I guess the problem comes from the parametrization of the cloudflare provider into LEGO 👍
The TTL and the check timeout are equals (120s) that's why even if we try twice the problem persists.
```

### What did you expect to see?

Certificates being generated

### What did you see instead?

`Error 403 - urn:ietf:params:acme:error:unauthorized - Incorrect TXT record ... found at _acme-challenge.domain.eu.com`

### Output of `traefik version`: (_What version of Traefik are you using?_)

<details>
  <summary>Version Info</summary>
  
  ```
  Version:      v1.7.0-rc3
  Codename:     maroilles
  Go version:   go1.10.3
  Built:        2018-08-01_01:37:51PM
  OS/Arch:      linux/amd64
  ```
</details>

### What is your environment & configuration

Provider: cloudflare

<details>
  <summary>Config</summary>

  ```json
  {
  "LifeCycle": {
    "RequestAcceptGraceTimeout": 0,
    "GraceTimeOut": 0
  },
  "GraceTimeOut": 0,
  "Debug": true,
  "CheckNewVersion": true,
  "SendAnonymousUsage": false,
  "AccessLogsFile": "",
  "AccessLog": null,
  "TraefikLogsFile": "",
  "TraefikLog": null,
  "Tracing": null,
  "LogLevel": "INFO",
  "EntryPoints": {
    "http": {
    "Address": "xxxx",
    "TLS": null,
    "Redirect": {
      "entryPoint": "xxxx"
    },
    "Auth": null,
    "WhitelistSourceRange": null,
    "WhiteList": null,
    "Compress": false,
    "ProxyProtocol": null,
    "ForwardedHeaders": null
    },
    "https": {
    "Address": "xxxx",
    "TLS": {
      "MinVersion": "",
      "CipherSuites": null,
      "Certificates": [],
      "ClientCAFiles": null,
      "ClientCA": {
      "Files": null,
      "Optional": false
      },
      "DefaultCertificate": null,
      "SniStrict": false
    },
    "Redirect": {
      "regex": "xxxx",
      "replacement": "xxxx"
    },
    "Auth": null,
    "WhitelistSourceRange": null,
    "WhiteList": null,
    "Compress": false,
    "ProxyProtocol": null,
    "ForwardedHeaders": null
    }
  },
  "Cluster": null,
  "Constraints": [],
  "ACME": {
    "Email": "xxxx",
    "Domains": [],
    "Storage": "xxxx",
    "StorageFile": "",
    "OnDemand": false,
    "OnHostRule": false,
    "CAServer": "",
    "EntryPoint": "xxxx",
    "KeyType": "",
    "DNSChallenge": null,
    "HTTPChallenge": null,
    "TLSChallenge": null,
    "DNSProvider": "",
    "DelayDontCheckDNS": 0,
    "ACMELogging": false,
    "OverrideCertificates": false,
    "TLSConfig": null
  },
  "DefaultEntryPoints": [
    "https",
    "http"
  ],
  "ProvidersThrottleDuration": 2000000000,
  "MaxIdleConnsPerHost": 200,
  "IdleTimeout": 0,
  "InsecureSkipVerify": false,
  "RootCAs": null,
  "Retry": {
    "Attempts": 0
  },
  "HealthCheck": {
    "Interval": 30000000000
  },
  "RespondingTimeouts": null,
  "ForwardingTimeouts": null,
  "AllowMinWeightZero": false,
  "Web": null,
  "Docker": {
    "Watch": true,
    "Filename": "",
    "Constraints": null,
    "Trace": false,
    "TemplateVersion": 0,
    "DebugLogGeneratedTemplate": false,
    "Endpoint": "xxxx",
    "Domain": "",
    "TLS": null,
    "ExposedByDefault": false,
    "UseBindPortIP": false,
    "SwarmMode": false,
    "Network": ""
  },
  "File": null,
  "Marathon": null,
  "Consul": null,
  "ConsulCatalog": null,
  "Etcd": null,
  "Zookeeper": null,
  "Boltdb": null,
  "Kubernetes": null,
  "Mesos": null,
  "Eureka": null,
  "ECS": null,
  "Rancher": null,
  "DynamoDB": null,
  "ServiceFabric": null,
  "Rest": null,
  "API": {
    "EntryPoint": "traefik",
    "Dashboard": true,
    "Debug": false,
    "CurrentConfigurations": null,
    "Statistics": null,
    "DashboardAssets": null
  },
  "Metrics": null,
  "Ping": {
    "EntryPoint": "traefik"
  },
  "HostResolver": null,
  "ConfigFile": "/traefik.toml"
  }
  ```
</details>

### Logs

[On failure](https://github.com/containous/traefik/files/2290429/failure.log)

[On success](https://github.com/containous/traefik/files/2290432/success.log)

**Comment by nmengin** [comment]
Hello @A-Shleifman ,

Many thanks for opening the issue and for your interest in the porject.

A Pull Request which is already opened on LEGO seems to solve all problems with wildcard and base name domains.

I just proposed a modification in this way on the PR: https://github.com/xenolf/lego/pull/607#issuecomment-414360798

I have some tests to make with the modifications but for the moment the PR (+ my modifications) seems to work.
If you want, you can make your own tests with my _home-made_ Træfik image `nmengin/traefik:acme-dns`.

If the tests are OK and  if the PR is validated by the LEGO team we will update the dependencies ASAP in the way to integrate its modifications.

**Comment by nmengin** [comment]
Hello @A-Shleifman ,

I confirm that your problem is solved with the LEGO PR. I just tested it sucessfully with cloudflare.

**Comment by A-Shleifman** [comment]
Hi @nmengin,

Thanks a lot for the reply and for the modification!

Just tested it with your image as well and it works perfectly. Traefik retrieved certificates in a few seconds :)

**Comment by traefiker** [comment]
Closed by #3888.
